CharlieBackend.Business/Services/StudentGroupService.cs

```diff
@@ -229,17 +229,35 @@ public async Task<Result<IList<StudentGroupDto>>> GetAllStudentGroupsAsync(DateT
         {
             if (!(startDate is null) || !(finishDate is null))
             {
-                return await GetStudentGroupsByDateAsyns(startDate, finishDate);
+                return await GetStudentGroupsByDateAsync(startDate, finishDate);
             }
-            var studentGroup = await _unitOfWork.StudentGroupRepository.GetAllAsync();
-            var studentGroupDto = _mapper.Map<List<StudentGroupDto>>(studentGroup);
+            var allStudentGroups = await _unitOfWork.StudentGroupRepository.GetAllAsync();
```

Comment:
Please implement GetAllActive() method in the repository and use it instead of GetAllAsync() + HideUnactiveStudentGroups